spec/monotonic_tick_count_spec.rb

```diff
@@ -129,11 +129,22 @@
 
   context "class methods" do
     context ".now" do
-      it "return an instance containing the current global monotonic tick counter" do
+      it "should return an instance containing the current global monotonic tick counter" do
         expect(Process).to receive(:clock_gettime).with(Process::CLOCK_MONOTONIC) { 1234.012345 }
         tick_count = MonotonicTickCount.now
         expect(tick_count.tick_count_f).to eq(1234.012345)
       end
     end
+
+    context ".timer" do
+      it "should return the result and elapsed ticks of the given block" do
+        result, duration = MonotonicTickCount.timer do
+          sleep(2.718)
+          1
+        end
+        expect(result).to be == 1
+        expect(duration).to be_within(0.01).of(2.718)
```

Comment:
I like that this test doesn't use mocking, but we've found it to be pretty pretty to count on timers running precisely. In practice, when there are other tenants on the hardware, it could take longer than the sleep time. You could widen the range up a lot, but it's hard to know how many seconds of slop you might need? At some point the slop is so large, the test isn't actually testing much. It's probably best to just mock the `Process` tick count method to return two values that differ by the expected amount.

Reply:
Understandable. I know that the general guideline is to avoid mocks when possible. But timing seems like a good exception to the rule.

Reply:
Yep, we should definitely avoid mocks when we can. But they are pretty necessary when there's a dependency on a process global function (the case here) or an external process or service that won't be running when our test is.